### ISSUE TITLE
FSharp.Core included in code coverage

### ISSUE THREAD

**Issue by halcwb** [TheAngryByrd/MiniScaffold#210]
**Describe the bug**
When upgrading to FSharp.Core 4.7.1, running DotnetTest results in FSharp.Core being included in the code coverage.

**To Reproduce**
1. Create a MiniScaffold test sample.
2. Upgrade FSharp.Core to 4.7.1 in the dependencies file
3. Install with dotnet paket install
4. Run the tests with ./build DotnetTest

**Expected behavior**
Code coverage should only include the MiniScaffold tests.

Instead you get this:

>  error : Coverage percentage achieved is 71% below the threshold of 80%.

This is because FSharp.Core is included:

```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<CoverageSession xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <Summary numSequencePoints="13892" visitedSequencePoints="1301" numBranchPoints="12740" visitedBranchPoints="591" sequenceCoverage="9.37" branchCoverage="4.64" maxCyclomaticComplexity="498" minCyclomaticComplexity="1" visitedClasses="195" numClasses="2421" visitedMethods="514" numMethods="5348" minCrapScore="1" maxCrapScore="2314.41" />
  <Modules>
    <Module hash="8C-CB-BC-98-32-88-EC-B9-F9-C1-26-19-6F-6C-5B-4A-1C-C7-47-B2">
      <Summary numSequencePoints="13885" visitedSequencePoints="1294" numBranchPoints="12733" visitedBranchPoints="591" sequenceCoverage="9.32" branchCoverage="4.64" maxCyclomaticComplexity="498" minCyclomaticComplexity="1" visitedClasses="190" numClasses="2416" visitedMethods="507" numMethods="5341" minCrapScore="1" maxCrapScore="2314.41" />
      <ModulePath>C:\Development\lab\MiniScaffold\tests\MiniScaffold.Tests\bin\Debug\netcoreapp3.1\__Saved\FSharp.Core.dll</ModulePath>
      <ModuleTime>2020-02-04T22:25:18Z</ModuleTime>
      <ModuleName>FSharp.Core</ModuleName>
      <Files>
        <File uid="1" fullPath="F:\workspace\_work\1\s\src\fsharp\FSharp.Core\prim-types.fs" />
        <File uid="2" fullPath="F:\workspace\_work\1\s\src\fsharp\FSharp.Core\seqcore.fs" />
        <File uid="3" fullPath="F:\workspace\_work\1\s\src\fsharp\FSharp.Core\seq.fs" />


```

**Comment by TheAngryByrd** [comment]
This sounds like an issue that should be opened in the [AltCover repo](https://github.com/SteveGilham/altcover).

**Comment by halcwb** [comment]
I fixed by:

```
let dotnetTest ctx =
    let excludeCoverage =
        !! testsGlob
        |> Seq.map IO.Path.GetFileNameWithoutExtension
        |> Seq.append ["FSharp.Core"]
        |> String.concat "|"
        
    let args =
        [
            "--no-build"
            "/p:AltCoverForce=true"
            sprintf "/p:AltCover=%b" (not disableCodeCoverage)
            sprintf "/p:AltCoverThreshold=%d" coverageThresholdPercent
            sprintf "/p:AltCoverAssemblyExcludeFilter=%s" excludeCoverage
        ]
    DotNet.test(fun c ->

        { c with
            Configuration = configuration (ctx.Context.AllExecutingTargets)
            Common =
                c.Common
                |> DotNet.Options.withAdditionalArgs args
            }) sln

```
So I added FSharp.Core to the AltCoverAssemblyExcludeFilter and force the intermediate __Saved dir to be deleted. This solves my problems.

**Comment by TheAngryByrd** [comment]
Yeah I feel like this is an issue that AltCover should handle since I could see how this might extend to all dependent libraries. Or maybe we need to be explicit about the libraries we do include. Hmm

**Comment by baronfel** [comment]
Feels like altcover could collect the -r references (since it's being run as part of a build?) and implicitly use them as the exclude filter...

**Comment by simontaite** [comment]
You can add this to the arguments to force AltCover to only consider code that it has access to the source code for
`/p:AltCoverLocalSource=true`

```fsharp
let args =
        [
            "--no-build"
            sprintf "/p:AltCover=%b" (not disableCodeCoverage)
            sprintf "/p:AltCoverThreshold=%d" coverageThresholdPercent
            sprintf "/p:AltCoverAssemblyExcludeFilter=%s" excludeCoverage
            "/p:AltCoverLocalSource=true"
        ]
```

**Comment by TheAngryByrd** [comment]
Closed by #218
